src/osd/OSD.cc

```diff
@@ -5719,6 +5719,8 @@ void OSD::send_full_update()
     state = CEPH_OSD_BACKFILLFULL;
   } else if (service.is_nearfull()) {
     state = CEPH_OSD_NEARFULL;
+  } else {
```

Comment:
this is wrong, the new state will reset the previous full state if any. so even a `0` MOSDFull is meaningful to mon.